src/plugins/julia/templates/lr.template.jl

```diff
@@ -0,0 +1,243 @@
+module SyntaxParser
+
+#=
+  LR parser generated by the Syntax tool.
+
+  https://www.npmjs.com/package/syntax-cli
+
+    npm install -g syntax-cli
+
+    syntax-cli --help
+
+  To regenerate run:
+
+    syntax-cli \
+      --grammar ~/path-to-grammar-file \
+      --mode <parsing-mode> \
+      --output ~/ParserClassName.jl
+=#
+
+# --------------------------------------------------------------
+# Exports: The parse method, if concerned about disambiguating with other
+# parse methods in your use case, use 'import' instead of 'using'
+export parse
+
+# --------------------------------------------------------------
+# Shared includes
+using DataStructures
+
+# Constants and globals
+const EOF = "\$"
+yytext = ""
+yylength = 0
+__res = nothing
+__loc = nothing
+should_capture_locations = {{{CAPTURE_LOCATIONS}}}
+
+# Types
+struct SyntaxError <: Exception
+    msg::AbstractString
+end
+
+function Base.showerror(io::IO, err::SyntaxError)
+    print(io, err.msg)
+end
+
+Base.@kwdef mutable struct yyLoc
+    startoffset
```

Comment:
Does Julia have recommended way of naming fields/variables with multiple words? E.g. underscore_case for `start_offset` or camelCase for `startOffset`? Ditto for other variables/fields in these files.

Reply:
This is a Julia oddity I came across - here's the style guide, refer to the section on "Use naming conventions consistent with Julia Base" which includes the note "modules and type names use capitalization and camel case: module SparseArrays, struct UnitRange" followed by "functions are lowercase and, when readable, with multiple words squashed together" - https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/ - this squash together bit seems to be consistently used for values as well within a type, such as variables defined in a struct or function.